templates/configmaps_st2-conf.yaml

```diff
@@ -27,7 +27,7 @@ data:
     url = amqp://{{ required "rabbitmq-ha.rabbitmqUsername is required!" (index .Values "rabbitmq-ha" "rabbitmqUsername") }}:{{ required "rabbitmq-ha.rabbitmqPassword is required!" (index .Values "rabbitmq-ha" "rabbitmqPassword") }}@{{ .Release.Name }}-rabbitmq-ha-discovery:5672
     [database]
     # TODO: ReplicaSet connection string needs templating based on number of MongoDB nodes
-    host = mongodb://{{ .Release.Name }}-mongodb-ha/?replicaSet=rs0
+    host = mongodb://{{ required "mongodb-ha.auth.adminUser is required!" (index .Values "mongodb-ha" "auth" "adminUser") }}:{{ required "mongodb-ha.auth.adminPassword is required!" (index .Values "mongodb-ha" "auth" "adminPassword") }}@{{ .Release.Name }}-mongodb-ha/?authSource=admin&replicaSet=rs0
```

Comment:
Per @armab, add a condition and set mongo username/password in `st2.conf` only if `mongodb-ha.auth.enabled = true`.

Reply:
Done